lib/instrumentation/index.js

```diff
@@ -209,6 +213,11 @@ Instrumentation.prototype.start = function (runContextClass) {
 
   this._runCtxMgr.enable()
   this._startHook()
+
+  if (nodeHasInstrumentableFetch && this._isModuleEnabled('undici')) {
+    this._log.debug('instrumenting fetch')
+    undiciInstr.instrumentUndici(this._agent)
```

Comment:
(I'm afraid to ask but) -- what's driving the need for this manual call to `instrumentUndici`.  Naively -- it seems like adding undichi to `MODULES` will always load `lib/instrumentation/modules/undici.js` which also calls `instrumentUndici`.  What's the weird case that leads us to  need this second, earlier, call to `instrumentUndici`?

Reply:
> adding undichi to MODULES will always load lib/instrumentation/modules/undici.js 

It will load `lib/instrumentation/modules/undici.js`, yes, but the shimming (`shimUndici`) isn't called unless the require-in-the-middle hook sees a `require('undici')`.

> What's the weird case that leads ...

To instrument node's core `fetch()` implementation we need to enable the undici instrumentation **even if there is no `require('undici')`** anywhere. This is because node's core `fetch()` uses undici as an esbuild bundle: https://github.com/nodejs/node/blob/main/deps/undici/undici.js
That we can even instrument that is down to the hooks being via diagnostics_channel rather than needing to monkey-patch.

And to avoid double-instrumenting, there is a guard at the start of `instrumentUndici`:

```js
function instrumentUndici (agent) {
  if (isInstrumented) {
    return
  }
  isInstrumented = true
...
```


> I'm not sure if the lack of a CHANGELOG entry is deliberate or just an oversight.

Oh, that is totally an oversight. Thanks for pointing that out!